### PR TITLE
run conformance tests on three Kubernetes versions

### LINKED ISSUE #493: test EG against multiple Kubernetes versions
We should test EG against multiple Kubernetes versions -- IMO at least the latest 3 minor versions.

### PR DESCRIPTION
**Pull request by skriss**
Closes #493.

Signed-off-by: Steve Kriss <krisss@vmware.com>

### CODE REVIEW

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -52,16 +50,33 @@ jobs:
         files: ./coverage.xml
         name: codecov-envoy-gateway
         verbose: true
+  
+  conformance-tests:
+    runs-on: ubuntu-latest
+    needs: [unit-tests]
+    strategy:
+      matrix:
+        kind_node_tag: ["v1.24.0", "v1.23.6", "v1.22.9"]
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:59]: this will increase PR time by 6min, I think thats fine for now

Comment (skriss) [reply .github/workflows/build_and_test.yaml:59]: They run in parallel though

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -52,16 +50,33 @@ jobs:
...
+
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    needs: [conformance-tests]
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./tools/github-actions/setup-deps
 
-    # build
     - name: Build Multiarch EG Binaries
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:77]: I think order matters here, we build the binary and then copy it into the image

File: .github/workflows/build_and_test.yaml (hunk 1 above)
Comment (arkodg) [reply .github/workflows/build_and_test.yaml:59]: ah amazing !

File: .github/workflows/build_and_test.yaml (hunk 2 above)
Comment (skriss) [reply .github/workflows/build_and_test.yaml:77]: I'm pretty sure that `make conformance` ends up rebuilding the binary even if it already exists locally, but I'll have to check again tmrw.

Comment (arkodg) [reply .github/workflows/build_and_test.yaml:77]: nvm it should be fine

File: .github/workflows/build_and_test.yaml [hunk 3]
```diff
@@ -52,16 +50,33 @@ jobs:
         files: ./coverage.xml
         name: codecov-envoy-gateway
         verbose: true
+  
+  conformance-tests:
+    runs-on: ubuntu-latest
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:55]: we ideally want unit-tests, conformance and push to be on the same runner

Comment (skriss) [reply .github/workflows/build_and_test.yaml:55]: AFAIK there's no way to run steps within a job in parallel, hence splitting conformance out into its own matrix job so it's not too slow. We can use upload/download artifact steps to share the binaries between jobs (this is the gist of https://github.com/envoyproxy/gateway/issues/683). Are there other reasons for wanting everything on the same runner?

Comment (arkodg) [reply .github/workflows/build_and_test.yaml:55]: the only one is to make sure the same binary is used between all steps, so recommend
* running the jobs sequentially in this PR and then
* #683 can be used to enhance the CI to be able to share builds across runners

**Review by arkodg — APPROVED** [review]

**Pull request merged**